test/srv6-inject/src/nf.rs

```diff
@@ -1,11 +1,43 @@
 use colored::*;
 use generic_array::typenum::*;
+use generic_array::GenericArray;
 use netbricks::headers::*;
+use netbricks::interface::*;
 use netbricks::operators::*;
 use netbricks::scheduler::Scheduler;
 use std::net::Ipv6Addr;
 use std::str::FromStr;
 
+macro_rules! srh_into_packet_gen {
+    ($segments:expr, $pkt:expr, $meta:expr, [$($len:expr => $ty:ty),*]) => {
+        match $segments.len() {
+            $(
+                $len => {
+                    let mut srh = <SegmentRoutingHeader<Ipv6Header, $ty>>::new(
+                        *GenericArray::<_, $ty>::from_slice(&$segments[..]),
+                    );
+
+                    let (nh, payload_len) = $meta;
+
+                    srh.set_segments_left(1);
+                    srh.set_last_entry(1);
+
+                    srh.ext_header.set_next_header(nh);
+
+                    $pkt.get_mut_header()
+                        .set_dst(Segment::from($segments[$segments.len() - 1]));
+                    $pkt.get_mut_header()
+                        .set_payload_len(payload_len + (srh.offset() as u16));
+                    $pkt.insert_header(&srh).unwrap();
```

Comment:
Macro Yo! Maybe we can bubble up the Result from here? and have the catch all arm also return Err of something? We can also just leave it be and come back when we figure out how to pretty it all up. 👍

Reply:
@drunkirishcoder the any case returns void... which we can then filter the original packet on, as I added.

Reply:
got it. and fixed.